### PR TITLE
Fallback on relation property name, for consistency

### PR DESCRIPTION
**Pull request by fquffio**
This PR adds a fallback on association's property name if no association with the exact name is found.

This solves an issue where `loadRelatedObjects(42, 'documents', 'has_poster')` wouldn't work, even if the `has_poster` relation is available for Documents, due to the fact that the association name is "HasPoster".

### CODE REVIEW

File: src/Model/ObjectsLoader.php [hunk 1]
```diff
@@ -187,10 +187,23 @@ public function hydrateObjects(array $objects): CollectionInterface
...
+        }
+
+        /*
+         * @todo The following piece of junk should be removed as soon as practically possible.
+         *      It's left here for backwards compatibility with a longstanding bug, but it's wrong.
+         */
+        $name = strtolower($name);
+        foreach ($associations as $assoc) {
+            if (strtolower($assoc->getName()) === $name) {
+                deprecationWarning('Using lowercased association name is a bug and will be removed soon: please use its correct name in CamelCase or the property name in snake_case');
```
Comment (le0m) [inline src/Model/ObjectsLoader.php:204]:
```suggestion
                deprecationWarning(sprintf('Using the lowercased association name "%s" is a bug and will be removed soon: please use its correct name in CamelCase or the property name in snake_case', $name));
```
Give a hint to which association caused the deprecation notice, even if `$name` has already been lowercased at this point.

Comment (fquffio) [reply src/Model/ObjectsLoader.php:204]: I see what you want here, but interpolation in a deprecation warning message isn't great in my opinion, as tools aim to aggregate such messages, and CakePHP itself disregards duplicate warning messages.

Interpolation would disrupt both aggregation _and_ deduplication. I'd rather trigger a warning _and_ log a message instead, where the original name could be included to make debugging easier.

Comment (le0m) [reply src/Model/ObjectsLoader.php:204]:
```suggestion
                deprecationWarning('Using lowercased association name is a bug and will be removed soon: please use its correct name in CamelCase or the property name in snake_case');
                Log::notice(sprintf('Using lowercased association name is a bug and will be removed soon: please use its correct name in CamelCase or the property name in snake_case. The wrong association name is: "%s"', $name));
```
Something like this?

Comment (fquffio) [reply src/Model/ObjectsLoader.php]: Done! Can you please check if the message I've used is clear enough?

Comment (le0m) [reply src/Model/ObjectsLoader.php]: lgtm

**Review by le0m — APPROVED** [review]

**Pull request merged**